### PR TITLE
only async trigger changes

### PR DESCRIPTION
**Pull request by hanshasselberg**
This is a tiny step towards fixing https://github.com/hashicorp/consul/issues/6616.

This PR stops all agent endpoints from syncing the changes synchronously and leaves it to the asynchronous ticker to pick up these changes. And makes changes to how `syncNodeInfo` is called...

The evolution of `syncNodeInfo`:

original:

1. `updateSyncState` -- among other things -- determines if node needs syncing and set `nodeInfoInSync`
2. `SyncChanges` adds node info to every service/check update if `!nodeInfoInSync` and has a manual `syncNodeInfo` call in case everything before failed to updated nodeinfo

after #7189:

1. `updateSyncState` determines if node needs syncing and set `nodeInfoInSync`
2. `SyncChanges` calls `syncNodeInfo` upfront instead of appending nodeinfo to every service/check update

proposed:

1. `updateSyncState` determines if node needs syncing and calls `syncNodeInfo` right away
2. `SyncChanges` doesn't try to sync node info

This PR potentially makes the tests more flaky because services/checks are no longer synced immediately.

### CODE REVIEW

File: api/health_test.go [hunk 1]
```diff
@@ -212,6 +212,16 @@ func TestAPI_HealthChecks(t *testing.T) {
 		t.Fatalf("err: %v", err)
 	}
 
+	retry.Run(t, func(r *retry.R) {
```
Comment (i0rek) [inline api/health_test.go:215]: Wait for service to sync to catalog. `testrpc` cannot be imported here.

File: testrpc/wait.go [hunk 2]
```diff
@@ -64,6 +64,7 @@ func WaitUntilNoLeader(t *testing.T, rpc rpcFn, dc string, options ...waitOption
 type waitOption struct {
 	Token                  string
 	WaitForAntiEntropySync bool
+	WaitForService         string
```
Comment (i0rek) [inline testrpc/wait.go:67]: Add ability to wait for a service because services are no longer synced immediately.

File: sdk/testutil/server.go [hunk 3]
```diff
@@ -250,7 +250,6 @@ func NewTestServerConfigT(t TestingTB, cb ServerConfigCallback) (*TestServer, er
 		return nil, errors.Wrap(err, "failed marshaling json")
 	}
 
-	t.Logf("CONFIG JSON: %s", string(b))
```
Comment (i0rek) [inline sdk/testutil/server.go:253]: remove debug.

File: agent/consul/server_test.go [hunk 4]
```diff
@@ -1031,8 +1031,8 @@ func TestServer_Leave(t *testing.T) {
 		t.Fatal("leave failed: ", err)
 	}
 
-	// Should lose a peer
-	retry.Run(t, func(r *retry.R) {
+	timer := &retry.Timer{Timeout: 10 * time.Second, Wait: 500 * time.Millisecond}
+	retry.RunWith(timer, t, func(r *retry.R) {
```
Comment (i0rek) [inline agent/consul/server_test.go:1035]: hopefully fixing a flaky test with this.

File: agent/local/state.go [hunk 5]
```diff
@@ -1341,22 +1317,17 @@ func (l *State) syncNodeInfo() error {
...
 		// todo(fs): some backoff strategy might be a better solution
-		l.nodeInfoInSync = true
 		accessorID := l.aclAccessorID(at)
 		l.logger.Warn("Node info update blocked by ACLs", "node", l.config.NodeID, "accessorID", accessorID)
 		metrics.IncrCounter([]string{"acl", "blocked", "node", "registration"}, 1)
-		return nil
 
 	default:
 		l.logger.Warn("Syncing node info failed.", "error", err)
-		return err
```
Comment (dnephin) [inline agent/local/state.go:1359]: What are the consequences of this function no longer returning an error?

**Review by dnephin — COMMENTED** [review]
Nice! I am not very familiar with state syncing, but I think this makes sense. Just one question about the change in error return

File: agent/local/state.go (hunk 5 above)
Comment (i0rek) [reply agent/local/state.go]: Based on your comment I revisited this part and reverted to returning and handling any errors from `syncNodeInfo`. I also added a longer explanation for these changes to the PR description.

**Comment by jmurret** [comment]
Closing as this is over two years old.

**Pull request closed without merging**